crates/anvil/src/eth/backend/fork.rs

```diff
@@ -674,8 +674,8 @@ pub struct ForkedStorage {
     pub geth_transaction_traces: HashMap<B256, GethTrace>,
     pub block_traces: HashMap<u64, Vec<Trace>>,
     pub block_receipts: HashMap<u64, Vec<TransactionReceipt>>,
-    pub eth_gas_estimations: HashMap<(Arc<CallRequest>, u64), U256>,
-    pub eth_call: HashMap<(Arc<CallRequest>, u64), Bytes>,
+    pub eth_gas_estimations: HashMap<(Arc<TransactionRequest>, u64), U256>,
+    pub eth_call: HashMap<(Arc<TransactionRequest>, u64), Bytes>,
     pub code_at: HashMap<(Address, u64), Bytes>,
 }
```

Comment:
@mattsse running into a bit of an issue here—`TransactionRequest` needs to be `Hash` to be used as a key, but we use `OtherFields` for any additional fields on the tx, which in turn uses a `serde_json::Value` for storing values, and it does not implement `Hash`.

I see two ways out of this:
- We do a manual hash implementation for `TransactionRequest`, and just `.to_string()` the `serde_json::Value`s. This sounds expensive, but its the simplest way.
- We could also restructure this cache to not take `TransactionRequest`s—do you have any idea how we could do so? for eth_call, we could maybe extract a tx hash, but not sure how to do so with `eth_gas_estimations`.

Reply:
can just derive Hash on TransactionRequest

Reply:
@mattsse we can't, `serde_json::Value` does not impl `Hash`

Reply:
-.-

Reply:
unless u have another idea?